homeassistant/components/shelly/light.py

```diff
@@ -20,11 +21,33 @@
 from .const import DATA_CONFIG_ENTRY, DOMAIN
 from .entity import ShellyBlockEntity
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up lights for device."""
     wrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id]
-    blocks = [block for block in wrapper.device.blocks if block.type == "light"]
+
+    blocks = []
+    for block in wrapper.device.blocks:
+        if block.type == "light":
+            blocks.append(block)
+        elif (
+            block.type == "relay"
+            and wrapper.device.settings["relays"][int(block.channel)].get(
+                "appliance_type"
+            )
+            == "light"
+        ):
+            blocks.append(block)
+            entity_reg = await hass.helpers.entity_registry.async_get_registry()
+            for entity in entity_reg.entities.values():
```

Comment:
We should put this logic inside `shelly/__init__.py`. Now we will do this twice for a Shelly 2.5 with 2 light appliance types?

Reply:
Each channel of the Shelly 2.5 can have a different appliance_type and _`_init__.py` doesn't have a per block approach so I think the check should stay in each platform.